SpatialGDK/Source/SpatialGDK/Private/Tests/SpatialView/EntityComponentTestUtils.h

```diff
@@ -0,0 +1,176 @@
+	Schema_SerializeToBuffer(Lhs, LhsBuffer.Get(), Length);
+	Schema_SerializeToBuffer(Rhs, RhsBuffer.Get(), Length);
+	return FMemory::Memcmp(LhsBuffer.Get(), RhsBuffer.Get(), Length) == 0;
+}
+
+/** Returns true if Lhs and Rhs have the same component ID and state. */
+inline bool CompareComponentData(const ComponentData& Lhs, const ComponentData& Rhs)
+{
+	if (Lhs.GetComponentId() != Rhs.GetComponentId())
+	{
+		return false;
+	}
+	return CompareSchemaObjects(Lhs.GetFields(), Rhs.GetFields());
+}
+
+/** Returns true if Lhs and Rhs have the same component ID and events. */
+inline bool CompareComponentUpdateEvents(const ComponentUpdate& Lhs, const ComponentUpdate& Rhs)
+{
+	if (Lhs.GetComponentId() != Rhs.GetComponentId())
+	{
+		return false;
+	}
+	return CompareSchemaObjects(Lhs.GetEvents(), Rhs.GetEvents());
+}
+
+/** Returns true if Lhs and Rhs have the same component ID and state. */
+inline bool CompareComponentUpdates(const ComponentUpdate& Lhs, const ComponentUpdate& Rhs)
+{
+	if (Lhs.GetComponentId() != Rhs.GetComponentId())
+	{
+		return false;
+	}
+	return CompareSchemaObjects(Lhs.GetFields(), Rhs.GetFields()) &&
+		CompareSchemaObjects(Lhs.GetEvents(), Rhs.GetEvents());
+}
+
+/** Returns true if Lhs and Rhs have the same entity ID, component ID, and state. */
+inline bool CompareEntityComponentData(const EntityComponentData& Lhs, const EntityComponentData& Rhs)
+{
+	if (Lhs.EntityId != Rhs.EntityId)
+	{
+		return false;
+	}
+	return CompareComponentData(Lhs.Data, Rhs.Data);
+}
+
+/** Returns true if Lhs and Rhs have the same entity ID, component ID, and events. */
+inline bool CompareEntityComponentUpdateEvents(const EntityComponentUpdate& Lhs, const EntityComponentUpdate& Rhs)
+{
+	if (Lhs.EntityId != Rhs.EntityId)
+	{
+		return false;
+	}
+	return CompareComponentUpdateEvents(Lhs.Update, Rhs.Update);
+}
+
+/** Returns true if Lhs and Rhs have the same entity ID, component ID, state, and events. */
+inline bool CompareEntityComponentUpdates(const EntityComponentUpdate& Lhs, const EntityComponentUpdate& Rhs)
+{
+	if (Lhs.EntityId != Rhs.EntityId)
+	{
+		return false;
+	}
+	return CompareComponentUpdates(Lhs.Update, Rhs.Update);
+}
+
+/** Returns true if Lhs and Rhs have the same ID, component ID, data, state and events. */
+inline bool CompareEntityComponentCompleteUpdates(const EntityComponentCompleteUpdate& Lhs, const EntityComponentCompleteUpdate& Rhs)
+{
+	if (Lhs.EntityId != Rhs.EntityId)
+	{
+		return false;
+	}
+	return CompareComponentData(Lhs.CompleteUpdate, Rhs.CompleteUpdate) && CompareComponentUpdateEvents(Lhs.Events, Rhs.Events);
+}
+
+inline bool CompareEntityComponentId(const EntityComponentId& Lhs, const EntityComponentId& Rhs)
+{
+	return Lhs == Rhs;
+}
+
+template<typename T, typename Predicate>
+bool AreEquivalent(const TArray<T>& Lhs, const TArray<T>& Rhs, Predicate&& Compare)
+{
+	if (Lhs.Num() != Rhs.Num())
+	{
+		return false;
+	}
+
+	for (int i = 0; i < Lhs.Num(); i++)
```

Comment:
I'm not sure what is going on here. I think this is just checking that it's a perumutation Num times.
I think this entirely function can be replaced by is_permutation.

Reply:
Oh wow, that ended up here by accident... fixing it now.